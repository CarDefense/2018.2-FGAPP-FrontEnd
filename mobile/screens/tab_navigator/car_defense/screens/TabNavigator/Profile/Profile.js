import React, { Component } from 'react';
import {
    StyleSheet,
    Text,
    View,
    Image,
    FlatList,
    ScrollView,
    RefreshControl,
    TouchableOpacity,
    Alert
} from 'react-native';
import { Icon } from 'native-base';
import { CAR_API } from '../const/Const'

export default class UserProfileView extends Component {

    constructor(props) {
        super(props);

        this.state = {
            refreshing: false
        };
    }

    async componentDidMount() {
        const { state } = this.props.navigation;
        var id = state.params ? (state.params.user.id ? state.params.user.id : state.params.user.user_id) : undefined;
        let link = CAR_API + '/car/?token=' + id

        return fetch(link)
            .then((response) => response.json())
            .then((responseJson) => {

                this.setState({
                    isLoading: false,
                    dataSource: responseJson,
                }, function () {

                });

            })
            .catch((error) => {
                console.error(error);
            });
    }

    _onPressButton = (plate) => {
        const { state } = this.props.navigation;
        var id = state.params ? (state.params.user.id ? state.params.user.id : state.params.user.user_id) : undefined;
        url = CAR_API + '/delete_car/'


        let car = JSON.stringify({
            id_token: id,
            plate: plate
        })

        fetch(url, {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: car
        }).then(response => { return response.json() }
        ).then(jsonResponse => {
            Alert.alert(jsonResponse)
        }).catch(error => {
            Alert.alert("Falha na conexão")
        })
    }

    _onRefresh = () => {
        this.setState({ refreshing: true });
        this.componentDidMount().then(() => {
            this.setState({ refreshing: false });
        });
    }

    registerCar = () => {
        const { state } = this.props.navigation;
        var user = state.params ? state.params.user : undefined;
        this.props.navigation.navigate('RegisterCar', { user: user })
    }

    render() {
        const { state } = this.props.navigation;
        var user = state.params ? (state.params.user.first_name ? state.params.user.first_name : state.params.user.username) : undefined;
        var source = state.params.user.picture.data.url ? {uri: state.params.user.picture.data.url } : undefined;


        return (
            <View style={{ backgroundColor: "#00ACC1", flex: 1 }}>
                <ScrollView
                    style={styles.item}
                    refreshControl={
                        <RefreshControl
                            refreshing={this.state.refreshing}
                            onRefresh={this._onRefresh}
                        />
                    }
                >
                    <View style={styles.headerContent}>
                        <Image style={styles.avatar}
                            source={source}/>

                        <Text style={styles.name}>Olá, {user}! Tudo bem?</Text>
                        {/* <Text style={styles.userInfo}>jhonnydoe@mail.com </Text> */}
                        {/* <Text style={styles.userInfo}>Florida </Text> */}
                        <Text style={styles.userInfo}>Aqui estão os seus carros</Text>
                    </View>

                    <View>
                        <FlatList
                            data={this.state.dataSource}
                            renderItem={({ item }) => {
                                return (
                                    <View style={styles.item2}>
                                        <Text style={styles.text1}>{item.plate}</Text>
                                        <Text style={styles.text2}>Modelo: {item.model}   Cor: {item.color}</Text>
                                        <Icon
                                            type='FontAwesome'
                                            name="car"
                                            style={styles.icon1}
                                        />
                                        <TouchableOpacity
                                            color="grey"
                                            onPress={() => { this._onPressButton(item.plate) }}
                                            containerViewStyle={{ width: '10%' }}
                                        >
                                            <Icon
                                                type='FontAwesome'
                                                name="trash"
                                                style={{ color: "red", fontSize: 15 }}
                                            />
                                        </TouchableOpacity>
                                    </View>

                                );
                            }}
                            keyExtractor={({ id }, index) => id.toString()}
                        />
                    </View>
                    <View style={styles.container}>
                        <TouchableOpacity
                            style={styles.button}
                            onPress={() => this.registerCar()}
                            containerViewStyle={{ width: '40%' }}
                        >
                            <Text style={styles.text} >+</Text>

                        </TouchableOpacity>
                    </View>
                </ScrollView>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center'
    },
    text: {
        color: "white",
        fontSize: 30,
    },
    button: {
        backgroundColor: "#B2EBF2",
        borderRadius: 20,
        height: 40,
        width: 40,
        justifyContent: 'center',
        alignItems: 'center'
    },
    headerContent: {
        padding: 30,
        alignItems: 'center',
    },
    avatar: {
        width: 100,
        height: 100,
        borderRadius: 40,
        borderWidth: 5,
        borderColor: "white",
        marginBottom: 10,
    },
    name: {
        fontSize: 22,
        color: "white",
        fontWeight: '800',
    },
    userInfo: {
        fontSize: 15,
        color: "white",
        fontWeight: '800',
        marginTop: 30,
        marginBottom: 15
    },
    item: {
        marginTop: 20,
        marginBottom: 5
    },
    icon: {
        width: 30,
        height: 30,
        marginTop: 20,
    },
    info: {
        fontSize: 18,
        marginTop: 20,
        color: "#FFFFFF",
    },
    item2: {
        alignItems: "center",
        backgroundColor: "white",
        flexGrow: 1,
        padding: 20,
        borderRadius: 15,
        elevation: 4,
        margin: 25,
        marginTop: 2,

    },
    icon1: {
        color: "#26C6DA",
        fontWeight: '800',
        fontSize: 44,
        position: 'absolute',
        left: 10,
        marginTop: 5,
        bottom: 25// Keep some space between your left border and Image

    },
    text1: {
        color: "#26C6DA",
        fontWeight: '800',
        fontSize: 30
    },
    text2: {
        color: "#B2EBF2",
        fontWeight: '800',
        fontSize: 12
    }
});