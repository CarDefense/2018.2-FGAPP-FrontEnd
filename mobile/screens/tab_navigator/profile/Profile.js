import React, { Component } from "react";
import {
    View,
    Text,
    StyleSheet,
    Button,
} from "react-native";

class Profile extends Component {
  constructor(props) {
      super(props);
      this.state = {
        url:'http://'+process.env.REACT_NATIVE_PACKAGER_HOSTNAME+':8000',      
      }
  }

  _onPressButton = async () => {
      // Coloque seu ip aqui
      fetch(this.state.url+'/api/rest-auth/logout/', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
      }),
  })
  .then((response) => response.json())
  .then((responseJson) => {
    console.log(JSON.stringify(responseJson.detail));
    if(responseJson.detail=="Successfully logged out."){
      console.log(JSON.stringify('Log OUT'));
      this.props.navigation.navigate('WelcomeScreen')
    }
  })}
    render() {
        return (
            <View style={styles.container}>
            <Button
                  onPress={this._onPressButton}
                  title="Sair"
            />
            </View>
        );
    }
}
export default Profile;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center'
    }
});
