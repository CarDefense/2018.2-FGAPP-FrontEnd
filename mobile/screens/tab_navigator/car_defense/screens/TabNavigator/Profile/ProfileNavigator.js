import React, { Component } from "react";
import {StackNavigator} from 'react-navigation'

import Profile from './Profile'


export default class ProfileNavigator extends Component {

    render() {
        return (
            <ProfileStackNavigator />
        );
    }
}

const ProfileStackNavigator = new StackNavigator({

    Profile: {
        screen: Profile,
        navigationOptions: ({ navigation }) => ({
            header: null,
        }),
    },
})