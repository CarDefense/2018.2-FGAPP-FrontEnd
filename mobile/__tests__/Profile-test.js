import Profile from '../screens/tab_navigator/car_defense/screens/TabNavigator/Profile/Profile'
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

test('Profile snapShot', () => {
    const snap = renderer.create(
        <Profile/>
    ).toJSON()
    expect(snap).toMatchSnapshot()
})