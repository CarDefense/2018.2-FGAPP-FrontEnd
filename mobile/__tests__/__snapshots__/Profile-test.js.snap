// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00ACC1",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    style={
      Object {
        "marginBottom": 5,
        "marginTop": 20,
      }
    }
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "padding": 30,
          }
        }
      >
        <Image
          source={null}
          style={
            Object {
              "borderColor": "white",
              "borderRadius": 40,
              "borderWidth": 5,
              "height": 100,
              "marginBottom": 10,
              "width": 100,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 22,
              "fontWeight": "800",
            }
          }
        >
          Olá, 
          
          ! Tudo bem?
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "fontWeight": "800",
              "marginBottom": 15,
              "marginTop": 30,
            }
          }
        >
          Aqui estão os seus carros
        </Text>
      </View>
      <View>
        <RCTScrollView
          data={undefined}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          invertStickyHeaders={undefined}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#B2EBF2",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "opacity": 1,
              "width": 40,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            +
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
